src/renderers/WebGLRenderer.js

```diff
@@ -1242,6 +1242,20 @@ class WebGLRenderer {
 
 				currentRenderState = renderStateStack[ renderStateStack.length - 1 ];
 
+				// restore clipping uniforms and viewport
+
+				const renderStateCamera = currentRenderState.state.camera;
+
+				if ( _clippingEnabled === true ) clipping.setGlobalState( _this.clippingPlanes, renderStateCamera );
+
+				const viewport = renderStateCamera.viewport;
```

Comment:
The `.viewport` is present on the (sub)cameras of the XR camera, whereas the render state stores the XR camera itself. So I don't think this now actually restores the viewport properly.

I was thinking that it might make the whole viewport handling easier if the XR viewports are associated with the WebXR render target. That way switching back and forth between render targets should automatically restore the right viewport. So instead of passing a `viewport` into `renderScene` the viewport property of the WebXR render target would be updated for each eye.

Reply:
Good point! I leave the viewport related changes out for now.

Let's try your approach with a different PR.